### PR TITLE
Update KeyboardObserver

### PR DESCRIPTION
**Pull request by robmaceachern**
This updates the `KeyboardObserver` to match the more recent implementation we use elsewhere.

Changes include:
- Support for the iOS 16.1+ changes related to the screen coordinate space. This can impact reported values when the app isn't full screen in Split View, Slide Over, and Stage Manager.
- The delegate has been changed to report the `UIView.AnimationCurve` instead of `UIView.AnimationOptions`
- Various formatting changes to match other implementations.
- `ListView` changes:
  - We now use a `UIViewPropertyAnimator` instead of `UIView.animate` when responding to keyboard notifications.
  - The `KeyboardFrameWillChangeCallback` now receives the `UIView.AnimationCurve` instead of `UIView.AnimationOptions`

Note: This implementation 

### Checklist

Please do the following before merging:

- [x] Ensure any public-facing changes are reflected in the [changelog](https://github.com/kyleve/Listable/blob/main/CHANGELOG.md). Include them in the `Main` section.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,19 +2,20 @@
...
 - A new `isContentScrollable` property is added to `ListView` to determine if the content size is large enough that the list can be scrolled to a new position without springing back to it's original position.
-
 - A new `custom` case is added to `KeyboardAdjustmentMode` which allows the consumer to fully customize the inset behavior. `onKeyboardFrameWillChange` and `customScrollViewInsets` have been added to `ListView` and `updateScrollViewInsets` is now public. All of these can be utilized in conjunction to respond to the keyboard and fully control the insets. For now, these are available through `ListView` only.
 
 ### Removed
 
 ### Changed
 
 - `SwipeAction` property names have been updated to better reflect what they're for. `Completion` also now takes in a more descriptive enum, instead of a boolean, to make reading callsites clearer. Eg, `completion(.expandActions)` instead of `completion(true)`.
+- `KeyboardObserverDelegate` now provides `UIView.AnimationCurve` instead of `UIView.AnimationOptions`.
```
Comment (kyleve) [inline CHANGELOG.md:18]: God bless

File: ListableUI/Sources/KeyboardObserver/KeyboardObserver.swift [hunk 2]
```diff
@@ -1,10 +1,3 @@
-//
-//  KeyboardObserver.swift
-//  ListableUI
-//
-//  Created by Kyle Van Essen on 2/16/20.
```
Comment (kyleve) [inline ListableUI/Sources/KeyboardObserver/KeyboardObserver.swift:5]: Wow rude

File: ListableUI/Sources/KeyboardObserver/KeyboardObserver.swift [hunk 3]
```diff
@@ -200,27 +206,44 @@ public final class KeyboardObserver {
...
+    struct NotificationInfo: Equatable {
+
+        var endingFrame: CGRect = .zero
+
+        var animationDuration: Double = 0.0
+        var animationCurve: UIView.AnimationCurve = .easeInOut
 
-        var endingFrame : CGRect = .zero
+        @available(iOS 16.1, *)
+        var screen: UIScreen {
```
Comment (kyleve) [inline ListableUI/Sources/KeyboardObserver/KeyboardObserver.swift:229]: Is this intermediary useful? Can we not just do this in init?

**Review by kyleve — APPROVED** [review]
Couple small things, otherwise LGTM!

File: ListableUI/Sources/KeyboardObserver/KeyboardObserver.swift (hunk 3 above)
Comment (robmaceachern) [reply ListableUI/Sources/KeyboardObserver/KeyboardObserver.swift:229]: AFAIK, you can't have stored properties that are marked with `@available`. So to keep the public API as accurate as possible I'm using that private optional `UIScreen` property for storage. There might be a cleaner way to do this but I didn't come across anything when I looked.

Comment (kyleve) [reply ListableUI/Sources/KeyboardObserver/KeyboardObserver.swift:229]: A ha!

**Comment by robmaceachern** [comment]
@kyleve The notification screen handling was updated in https://github.com/square/Listable/pull/499/commits/27d578d368f8e02eed3f04a68ea2a923a3f5707e to match the revised Market implementation (https://github.com/squareup/market/pull/6826).

**Review by kyleve — APPROVED** [review]

**Pull request merged**